test/analysis/buildTypeWeights.test.ts

```diff
@@ -612,6 +734,55 @@ describe('Test buildTypeWeightsFromSchema function', () => {
                     },
                 });
             });
+
+            test('on union types', () => {
+                schema = buildSchema(`
+                union SearchResult = Human | Droid
+                type Human{
+                    name: String
```

Comment:
Can you add another field to both Human and Droid using the not-null operator? Somethling like `age: Int!`

This type should then appear in the list of fields for `searchresult`